# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediacontrol/MediaControlKeysEvent.h
# Commit: 714b2fb21e23
# Full Hash: 714b2fb21e2341e8e2bf1be1271ca0ed5e526443
# Author: alwu <alwu@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1640998
# File Overlap Count: 1
# Description:
#   Bug 1640998 - part5 : set supported keys to the event source r=chunmin
#   
#   This patch will
#   - add a method `SetSupportedMediaKeys()` on `MediaControlKeysEventSource`
#   - set main controller's supported key to the event source
# ==============================================================================

diff -r a48b6554ba9c -r 714b2fb21e23 dom/media/mediacontrol/MediaControlKeysEvent.h
--- a/dom/media/mediacontrol/MediaControlKeysEvent.h	Mon Jun 08 22:16:46 2020 +0000
+++ b/dom/media/mediacontrol/MediaControlKeysEvent.h	Tue Jun 09 02:23:35 2020 +0000
@@ -73,6 +73,8 @@
   NS_INLINE_DECL_PURE_VIRTUAL_REFCOUNTING
   MediaControlKeysEventSource();
 
+  using MediaKeysArray = nsTArray<MediaControlKeysEvent>;
+
   virtual void AddListener(MediaControlKeysEventListener* aListener);
   virtual void RemoveListener(MediaControlKeysEventListener* aListener);
   size_t GetListenersNum() const;
@@ -89,6 +91,10 @@
   // Override this method if the event source needs to handle the metadata.
   virtual void SetMediaMetadata(const MediaMetadataBase& aMetadata) {}
 
+  // Set the supported media keys which the event source can use to determine
+  // what kinds of buttons should be shown on the UI.
+  virtual void SetSupportedMediaKeys(const MediaKeysArray& aSupportedKeys) = 0;
+
  protected:
   virtual ~MediaControlKeysEventSource() = default;
   nsTArray<RefPtr<MediaControlKeysEventListener>> mListeners;